pkg/adapter/kubernetes/blueprintcr/v1/stateDiff.go

```diff
@@ -4,15 +4,20 @@ import (
 	"errors"
 	"fmt"
 	"github.com/cloudogu/k8s-blueprint-operator/pkg/domain"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/domain/common"
 )
 
 // StateDiff is the result of comparing the EffectiveBlueprint to the current cluster state.
 // It describes what operations need to be done to achieve the desired state of the blueprint.
 type StateDiff struct {
 	// DoguDiffs maps simple dogu names to the determined diff.
 	DoguDiffs map[string]DoguDiff `json:"doguDiffs,omitempty"`
-	// DoguDiffs maps simple dogu names to the determined diff.
+	// ComponentDiffs maps simple component names to the determined diff.
 	ComponentDiffs map[string]ComponentDiff `json:"componentDiffs,omitempty"`
+	// DoguConfigDiffs maps simple dogu names to the determined config diff.
+	DoguConfigDiffs map[string]CombinedDoguConfigDiff `json:"doguConfigDiffs,omitempty"`
+	// GlobalConfigDiff is the difference between the GlobalConfig in the EffectiveBlueprint and the cluster state.
+	GlobalConfigDiff GlobalConfigDiff `json:"globalConfigDiff,omitempty"`
```

Comment:
Name this field and type in plural too?

Reply:
Hm, I saw this on @alexander-dammeier's branch but I'm not really sure. In the end a diff can be a collection of as many diffs as you want. So it's not really clear when to use singular or plural.

Reply:
Okay. This isn't such a big thing anyway.